pcbdraw/pcbdraw.py

```diff
@@ -628,6 +661,53 @@ def component_from_library(lib, name, value, ref, pos, usedComponents, comp,
         }
         usedComponents[unique_name] = componentInfo
 
+        # If the library used is the THT resistor one, attempt to change the band colors if they exsist
+        if lib == "Resistor_THT":
+            res, tollerance = None, '5%'
+            try:
+                value = value.split(' ')
+                res = engineering_notation.EngNumber(value[0])
+                res = res.number
+                if len(value) >= 1:
+                    if '%' in value[1]:
+                        if value[1] not in style["tht-resistor-band-colors"]:
+                            raise UserWarning()
+                        tollerance = value[1]
+            except decimal.InvalidOperation:
+                if not silent:
+                    print("Resistor {}'s value is invalid".format(ref))
+            except UserWarning:
+                if not silent:
+                    print("Resistor {}'s tollerance ({}) is invalid, assuming 5%".format(ref, value[1]))
+            if res is not None:
+                power = math.floor(res.log10())-1
+                res = int(res / 10**power)
+                resistor_colors = [
+                    style["tht-resistor-band-colors"][int(str(res)[0])],
+                    style["tht-resistor-band-colors"][int(str(res)[1])],
+                    style["tht-resistor-band-colors"][int(power)],
+                    style["tht-resistor-band-colors"][tollerance],
+                ]
+                if tht_resistor_settings is not None:
+                    if ref in tht_resistor_settings:
+                        if 'flip' in tht_resistor_settings[ref]:
+                            if tht_resistor_settings[ref]['flip']:
+                                resistor_colors.reverse()
+
+                for res_i, res_c in enumerate(resistor_colors):
+                    band = componentElement.find(".//*[@id='{}res_band{}']".format(svg_prefix, res_i+1))
+                    s = band.attrib["style"].split(";")
+                    for i in range(len(s)):
+                        if s[i].startswith('fill:'):
+                            s_split = s[i].split(':')
+                            s_split[1] = res_c
+                            s[i] = ':'.join(s_split)
+                        elif s[i].startswith('opacity:'):
+                            s_split = s[i].split(':')
+                            s_split[1] = '1'
+                            s[i] = ':'.join(s_split)
+                    band.attrib["style"] = ";".join(s)
```

Comment:
Plese consider factoring out this code into a separate function. `component_from_library` is already too long and hard to follow.

Reply:
Yeah I agree. I split this into two functions: one that finds the resistor value and tolerance, and another to change the band's color and properties